sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -3364,7 +3364,12 @@ class Analyzer(override val catalogManager: CatalogManager) extends RuleExecutor
           leftKeys.map(_.withNullability(true)))
       case FullOuter =>
         // in full outer join, joinCols should be non-null if there is.
```

Comment:
This comment looks incorrect?

Reply:
I think the comment is technically correct, although hard to decipher maybe. I think "if there is" means "if there is non-null joinCols on some side". The coalesce is there to make that comment true.

I can make the comment more explicit.